### PR TITLE
BEI-14: Register users

### PR DESCRIPTION
**Pull request by hizikex**
#### WHY THIS PR?
To create a route for users to register on this platform.

#### WHAT DOES THIS PR DO?
This PR creates an API for users to register on foodies_haven with at least, their username, email and password.

#### HOW CAN THIS PR BE TESTED?
-This can be tested by inputing the wrong datatype for either or both username, email and password.
-By also input an already existing username or email.

#### HINGS YOU MAY WANT THE REVIEWER TO NOTE.
N/A

### CODE REVIEW

File: src/core/models/comments.ts [hunk 1]
```diff
@@ -34,15 +34,15 @@ Comment.init(
...
-  Comment.belongsTo(Recipe, {
-    foreignKey: 'recipe_id',
-    targetKey: 'id',
-    as: 'recipe',
-  });
+  // Comment.belongsTo(Recipe, {
+  //   foreignKey: 'recipe_id',
+  //   targetKey: 'id',
+  //   as: 'recipe',
+  // });
```
Comment (RICHYDOS) [inline src/core/models/comments.ts:48]: This should not be commented out

File: src/api/Route/user.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { Router } from 'express';
+import * as loginRequestHandler from '../requestHandlers/user';
```
[the remark this answers is not in the document]
Comment (hizikex) [reply src/api/Route/user.ts]: Done

File: src/api/Route/user.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+
+const router = Router();
+
+router.post('/register', loginRequestHandler.register);
```
[the remark this answers is not in the document]
Comment (hizikex) [reply src/api/Route/user.ts]: Okay

File: .gitignore [hunk 4]
```diff
@@ -2,3 +2,4 @@
 dist
 .env
 /coverage
+logs
```
Comment (RICHYDOS) [inline .gitignore:5]: Add an extra line

**Review by RICHYDOS — CHANGES_REQUESTED** [review]

File: src/core/controller/user.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+import UserRepository from "../repositories/User";
+import bcrypt from 'bcrypt';
+
+export const processRegistration = async (body: RegistrationRequest): Promise<UserRegisterResponse> => {
+    const userRepository = new UserRepository();
+    
+    const existingUser = await userRepository.findOne({ where: { email: body.email } });
+
+    if (existingUser) {
+        throw new Error('User with this email already exists.');
```
Comment (LucianaDera) [inline src/core/controller/user.ts:11]: Can we use a custom error, that throws a validation or conflict error

File: src/core/interfaces/user.ts [hunk 6]
```diff
@@ -8,3 +8,14 @@ export interface UserAttributes {
   created_at: Date;
   updated_at: Date;
 }
+
+export interface RegistrationRequest {
```
Comment (LucianaDera) [inline src/core/interfaces/user.ts:12]: UserRegisterationRequest

File: src/core/interfaces/user.ts [hunk 7]
```diff
@@ -8,3 +8,14 @@ export interface UserAttributes {
...
+  username: string;
+  email: string;
+  password: string;
+}
+
+export interface UserRegisterResponse {
```
Comment (LucianaDera) [inline src/core/interfaces/user.ts:18]: UserRegisterationResponse

File: src/core/models/followers.ts [hunk 8]
```diff
@@ -23,20 +23,18 @@ Follower.init(
       },
     {
       sequelize,
-      modelName: 'Follower',
-      timestamps: true,
+      tableName: 'followers',
```
Comment (LucianaDera) [inline src/core/models/followers.ts:26]: This shouldn't be part of the task

**Review by LucianaDera — CHANGES_REQUESTED** [review]

File: src/api/Route/index.ts [hunk 9]
```diff
@@ -1,11 +1,13 @@
 import { Router, Response, Request } from 'express';
-
+import { userRoutes } from './user';
 
 export const routes = (): Router => {
   const router = Router();
   router.get('/', (_req: Request, res: Response) => {
-    res.send('Welcome to the service template');
+    res.send('Welcome to foodies_haven');
```
[the remark this answers is not in the document]
Comment (hizikex) [reply src/api/Route/index.ts:7]: So what should I do here

File: .gitignore (hunk 4 above)
[the remark this answers is not in the document]
Comment (hizikex) [reply .gitignore]: There is already an extra line

**Review by LucianaDera — APPROVED** [review]

**Pull request merged**